### PR TITLE
Create a new store plugin

### PR DESCRIPTION
**Pull request by julien-lecomte**
##### New NWScript Functions
- Store: New `GetIsRestrictedBuyItem`
- Store: New `{Get|Set}BlackMarketMarkDown`
- Store: New `{Get|Set}MarkDown`
- Store: New `{Get|Set}MarkUp`

Since there's the possibilty of having a whitelist or blacklist for selling back to the store, I'm unsure if exposing just ``GetIsRestrictedBuyItem`` is sufficient.

### CODE REVIEW

File: Plugins/Store/Store.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    return -1;
+}
+
+NWNX_EXPORT ArgumentStack SetMarkUp(ArgumentStack&& args)
+{
+    if (auto *pStore = Utils::PopStore(args))
+    {
+        const auto nValue = args.extract<int32_t>();
+          ASSERT_OR_THROW(nValue >= 1);
+          ASSERT_OR_THROW(nValue <= 1000);
```
Comment (Daztek) [inline Plugins/Store/Store.cpp:77]: Why are the asserts `>=1` and `<= 1000` here while for the others it's `>=0` and `<=100`, that said, are they percentages or flat goldpiece values?

File: Plugins/Store/Store.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        pStore->m_nMarkUp = nValue;
+    }
+    return {};
+}
+
+NWNX_EXPORT ArgumentStack GetCustomerCount(ArgumentStack&& args)
+{
+
```
Comment (Daztek) [inline Plugins/Store/Store.cpp:85]: Style nit: please remove the extra newline

File: Plugins/Store/Store.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          ASSERT_OR_THROW(nValue >= 0);
+          ASSERT_OR_THROW(nValue <= 100);
+        pStore->m_nMarkDown = nValue;
+    }
+    return {};
+}
+
+NWNX_EXPORT ArgumentStack GetMarkUp(ArgumentStack&& args)
+{
+
```
Comment (Daztek) [inline Plugins/Store/Store.cpp:63]: Style nit: please remove the extra newline

File: Plugins/Store/Store.cpp (hunk 1 above)
Comment (julien-lecomte) [reply Plugins/Store/Store.cpp:77]: The values are the min and max that the toolset editor allows us to use in the UI. The're percentage points: 110 is 10% increase in price.

File: Plugins/Store/Store.cpp (hunk 2 above)
Comment (julien-lecomte) [reply Plugins/Store/Store.cpp]: Fixed.

File: Plugins/Store/Store.cpp (hunk 3 above)
Comment (julien-lecomte) [reply Plugins/Store/Store.cpp:63]: Fixed.

**Pull request merged**